atomate/common/firetasks/glue_tasks.py

```diff
@@ -73,6 +73,94 @@ def get_calc_loc(target_name, calc_locs):
         return calc_locs[-1]
 
 
+@explicit_serialize
+class GrabFilesFromCalcLoc(FiretaskBase):
+    """
+    Based on CopyVaspOutputs but for general file copying.
+
+    Required params:
+        filenames (str): filenames to copy.
+        name_prepend (str): string to prepend filenames, e.g. can be a directory.
+        name_append (str): string to append to filenames.
+
+    Optional params:
+        calc_dir: directory to copy from.
+        calc_loc: name of fw to get location for.
+    """
+
+    required_params = ["filenames", "name_prepend","name_append"]
+    optional_params = ["calc_dir", "calc_loc"]
+
+    def run_task(self,fw_spec=None):
+
+        if self.get('calc_dir',False):
+            filesystem = None
+        elif self.get('calc_loc',False):
+            calc_loc = get_calc_loc(self.get('calc_loc',False), fw_spec["calc_locs"])
+            calc_dir = calc_loc["path"]
+            filesystem = calc_loc["filesystem"]
+        else:
+            raise ValueError("Must specify either calc_dir or calc_loc!")
+
+        fileclient = FileClient(filesystem=filesystem)
+        calc_dir = fileclient.abspath(calc_dir)
+
+        all_files = fileclient.listdir(calc_dir)
+
+        if self.get('filenames',False):
```

Comment:
you don't need this if->else or *any* of self.get() statements

just refer to the variable as self["filenames"] directly. If the user didn't set the argument, Python will raise a pretty clear error by itself as to what went wrong. All the self.get(filename, False) should just become self["filenames"]